model/src/test/kotlin/config/ExcludesTest.kt

```diff
@@ -0,0 +1,207 @@
+            val projectId2 = Identifier("provider", "namespace", "project2", "version")
+
+            "return true if the package does not appear in the analyzer result" {
+                val excludes = Excludes()
+
+                val analyzerResult = AnalyzerResult(
+                        projects = sortedSetOf(),
+                        packages = sortedSetOf(),
+                        errors = sortedMapOf()
+                )
+
+                excludes.isPackageExcluded(id, analyzerResult) shouldBe true
+            }
+
+            "return true if all occurrences of the package are excluded" {
+                val excludes = Excludes(
+                        projects = listOf(ProjectExclude("path1", ProjectExcludeReason.BUILD_TOOL_OF, "comment")),
+                        scopes = listOf(ScopeExclude("testScope", ScopeExcludeReason.PROVIDED_BY, "comment"))
+                )
+
+                val analyzerResult = AnalyzerResult(
+                        projects = sortedSetOf(
+                                Project.EMPTY.copy(id = projectId1, definitionFilePath = "path1", scopes = sortedSetOf(
+                                        Scope("scope", sortedSetOf(PackageReference(id, sortedSetOf()))))),
+                                Project.EMPTY.copy(id = projectId2, definitionFilePath = "path2", scopes = sortedSetOf(
+                                        Scope("testScope", sortedSetOf(PackageReference(id, sortedSetOf())))
+                                ))
+                        ),
+                        packages = sortedSetOf(),
+                        errors = sortedMapOf()
+                )
+
+                excludes.isPackageExcluded(id, analyzerResult) shouldBe true
+            }
+
+            "return false if not all occurrences of the package are excluded" {
+                val excludes = Excludes(
+                        projects = listOf(ProjectExclude("path1", ProjectExcludeReason.BUILD_TOOL_OF, "comment")),
+                        scopes = listOf(ScopeExclude("testScope", ScopeExcludeReason.PROVIDED_BY, "comment"))
+                )
+
+                val analyzerResult = AnalyzerResult(
+                        projects = sortedSetOf(
+                                Project.EMPTY.copy(id = projectId1, definitionFilePath = "path1", scopes = sortedSetOf(
+                                        Scope("scope", sortedSetOf(PackageReference(id, sortedSetOf()))))),
+                                Project.EMPTY.copy(id = projectId2, definitionFilePath = "path2", scopes = sortedSetOf(
+                                        Scope("scope", sortedSetOf(PackageReference(id, sortedSetOf()))),
+                                        Scope("testScope", sortedSetOf(PackageReference(id, sortedSetOf())))
+                                ))
+                        ),
+                        packages = sortedSetOf(),
+                        errors = sortedMapOf()
+                )
```

Comment:
Almost all code from 196 to 177 is irrelevant for the test and it is the majority of code inside this test case.
How about stripping out all irrelevant stuff?